CMakeLists.txt

```diff
@@ -0,0 +1,165 @@
+
+find_package (BISON REQUIRED)
+
+
+# Compiler flags
+# --------------
+
+include_directories(
+    ${SWIG_SOURCE_DIR}/CParse
+    ${SWIG_SOURCE_DIR}/Include
+    ${SWIG_SOURCE_DIR}/DOH
+    ${SWIG_SOURCE_DIR}/Swig
+    ${SWIG_SOURCE_DIR}/Preprocessor
+    ${SWIG_SOURCE_DIR}/Modules
+    ${PROJECT_BINARY_DIR}/Source/Include
+    ${PROJECT_BINARY_DIR}/Source/CParse
+    ${PROJECT_SOURCE_DIR}/Source/Doxygen
+)
+
+# generate the parser source code (depends on bison)
+file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/Source/CParse)
+
+BISON_TARGET(swig_parser
+    ${SWIG_SOURCE_DIR}/CParse/parser.y
+    ${PROJECT_BINARY_DIR}/Source/CParse/parser.c
+)
+
+# generate swigwarn.swg
+file(READ ${SWIG_SOURCE_DIR}/Include/swigwarn.h SWIG_WARN_H)
+string(REGEX REPLACE "#define WARN([^ \\t]*)[ \\t]*([0-9]+)" "%define SWIGWARN\\1 \\2 %enddef" SWIG_WARN_SWG ${SWIG_WARN_H})
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/swigwarn.swg ${SWIG_WARN_SWG})
+set_property(SOURCE ${CMAKE_CURRENT_BINARY_DIR}/swigwarn.swg PROPERTY GENERATED 1)
+
+# install lib
+install (DIRECTORY ${SWIG_ROOT}/Lib/ DESTINATION ${SWIG_LIB})
+install (FILES ${CMAKE_CURRENT_BINARY_DIR}/swigwarn.swg DESTINATION ${SWIG_LIB})
+
+# Libraries
+# ---------
+
+file (GLOB CPARSE_SOURCES ${SWIG_SOURCE_DIR}/CParse/*.c)
+add_library (cparse ${CPARSE_SOURCES}
+    ${PROJECT_BINARY_DIR}/Source/CParse/parser.c
+    ${PROJECT_BINARY_DIR}/Source/CParse/parser.h
+)
+
+file (GLOB PREPROCESSOR_SOURCES ${SWIG_SOURCE_DIR}/Preprocessor/*.c)
+add_library (preprocessor ${PREPROCESSOR_SOURCES})
+
+file (GLOB DOH_SOURCES ${SWIG_SOURCE_DIR}/DOH/*.c)
+add_library (doh ${DOH_SOURCES})
+
+file (GLOB DOXYGEN_SOURCES ${SWIG_SOURCE_DIR}/Doxygen/*.cxx)
+add_library (doxygen ${DOXYGEN_SOURCES})
+
+file (GLOB CORE_SOURCES ${SWIG_SOURCE_DIR}/Swig/*.c)
+add_library (core ${CORE_SOURCES})
+if (PCRE_FOUND)
+  target_link_libraries (core ${PCRE_LIBRARIES})
+endif ()
+
+file (GLOB MODULES_SOURCES ${SWIG_SOURCE_DIR}/Modules/*.cxx)
+add_library (modules ${MODULES_SOURCES}
+    ${PROJECT_BINARY_DIR}/Source/Include/swigconfig.h
+    ${SWIG_SOURCE_DIR}/Include/swigwarn.h
+)
+target_link_libraries(modules doxygen)
+
+add_executable(swig
+    ${SWIG_SOURCE_DIR}/Modules/main.cxx
+    ${SWIG_SOURCE_DIR}/Modules/swigmain.cxx
+)
+
+target_link_libraries (swig cparse preprocessor doh core modules)
+install (TARGETS swig DESTINATION bin)
+
+
+# 'make package-source' creates tarballs
+set ( CPACK_PACKAGE_NAME ${PACKAGE_NAME} )
+set ( CPACK_SOURCE_GENERATOR "TGZ;TBZ2" )
+set ( CPACK_SOURCE_IGNORE_FILES "/.git;/build;.*~;${CPACK_SOURCE_IGNORE_FILES}" )
+set ( CPACK_SOURCE_PACKAGE_FILE_NAME ${PACKAGE_NAME}-${PACKAGE_VERSION} )
+include ( CPack )
+
+enable_testing()
+add_test (NAME cmd_version COMMAND swig -version)
+add_test (NAME cmd_external_runtime COMMAND swig -external-runtime ext_rt.h)
+set_tests_properties(cmd_external_runtime PROPERTIES ENVIRONMENT "SWIG_LIB=${PROJECT_SOURCE_DIR}/Lib")
```

Comment:
Unfortunately SWIG_LIB is complicated. Correct me if I'm wrong, but I think this is testing SWIG in the build tree, not where it is installed. Can you test it in the installed location easily? I think this is quite important wrt SWIG_LIB where SWIG_LIB must NOT be set when installed.

Reply:
Unfortunately we cannot check that the target was installed easily.